### PR TITLE
Update katello to user pulpcore 3.14, part 1

### PR DESCRIPTION
**Pull request by jjeffers**
Updates katello to use Pulpcore 3.14.

### CODE REVIEW

File: packages/katello/katello-repos/katello-repos.spec
Comment (ekohl) [inline packages/katello/katello-repos/katello-repos.spec:71]: You missed the release bump (line 8). Then this needs to match
```suggestion
* Thu Jul 2 2021 James Jeffers <jjeffers@redhat.com> - 4.2-0.2.nightly
```

**Review by ekohl — CHANGES_REQUESTED** [review]

File: packages/katello/katello-repos/katello-repos.spec
Comment (ehelms) [inline packages/katello/katello-repos/katello-repos.spec:71]: You are missing a bump to release on line 8 above.

Comment (evgeni) [reply packages/katello/katello-repos/katello-repos.spec:71]: I took the liberty to commit that, to get the builds further :)

**Comment by evgeni** [comment]
this PR deletes `packages/katello/rubygem-pulp_ansible_client/pulp_ansible_client-0.7.3.gem`, but I don't see a 0.8.0 version added, or the spec changed?

**Comment by evgeni** [comment]
:disappointed: the Ruby bindings seem to (again) contain the Python bindings too.

Justin had a crude fix for that in #6810 -- you might want to copy that over.

**Comment by evgeni** [comment]
in #6810 we also bumped pulpcore_client, deb_client and python_client -- I don't see these here -- intentional?

**Comment by evgeni** [comment]
And depends on something like https://github.com/Katello/katello/pull/9419 -- is there such a PR for 3.14 in the works?

**Comment by jjeffers** [comment]
> in #6810 we also bumped pulpcore_client, deb_client and python_client -- I don't see these here -- intentional?

From a few days ago, you had mentioned in another thread that pulp-python wasn't packaged yet. Would that exclude the python_client here?

**Comment by evgeni** [comment]
> > in #6810 we also bumped pulpcore_client, deb_client and python_client -- I don't see these here -- intentional?
> 
> From a few days ago, you had mentioned in another thread that pulp-python wasn't packaged yet. Would that exclude the python_client here?

The server (pulpcore) bits aren't packaged yet. But I see Katello depending on the client bindings, so I think these should be packaged/bumped.

**Comment by jjeffers** [comment]
> > > in #6810 we also bumped pulpcore_client, deb_client and python_client -- I don't see these here -- intentional?
> > 
> > 
> > From a few days ago, you had mentioned in another thread that pulp-python wasn't packaged yet. Would that exclude the python_client here?
> 
> The server (pulpcore) bits aren't packaged yet. But I see Katello depending on the client bindings, so I think these should be packaged/bumped.
> 
> Edit: But given we're not actually going to release this functionality just yet (due to the missing server bits), it doesn't need to be bumped in this PR.

Are packaging PRs squashed before merging?

**Comment by evgeni** [comment]
> Are packaging PRs squashed before merging?

usually not.

the python bindings have the same issue as the rpm ones: shipped python files :(

**Comment by jjeffers** [comment]
> > Are packaging PRs squashed before merging?
> 
> usually not.
> 
> the python bindings have the same issue as the rpm ones: shipped python files :(

Ok, I copied over the exclusion directives from Justin's abandoned pulp 3.13 update PR. I think I got them all.

**Comment by ianballou** [comment]
pulpcore-client 3.14.1(I'm guessing) will need to be released to deal with https://pulp.plan.io/issues/9008

**Comment by jjeffers** [comment]
> pulpcore-client 3.14.1(I'm guessing) will need to be released to deal with https://pulp.plan.io/issues/9008

Yup, waiting to see what the new gem versions will be before proceeding.

**Comment by ehelms** [comment]
Sounds like we are waiting on one thing before we merge this, in the meantime can you squash all of these commits together? I see early on you broke them out to individual update commits but now with all the changes piled on top that is muddy, so feels like better off squashing it down.

**Comment by jjeffers** [comment]
> Sounds like we are waiting on one thing before we merge this, in the meantime can you squash all of these commits together? I see early on you broke them out to individual update commits but now with all the changes piled on top that is muddy, so feels like better off squashing it down.

I will squash!

**Comment by jlsherrill** [comment]
3.14.1 is released, mind updating to that?  https://rubygems.org/gems/pulpcore_client/versions/3.14.1

**Comment by jlsherrill** [comment]
and now the final piece :)  https://rubygems.org/gems/pulp_rpm_client/versions/3.13.3

**Comment by evgeni** [comment]
Given only pulp_ansible_client and pulp_python_client have the problem of python code being included in the gem (probably because those releases don't have https://github.com/pulp/plugin_template/commit/36a8e7d73ea38bb4bf92ebb84d7e5263114b2fa6 applied yet), I would lean towards only applying the workaround there?

**Comment by evgeni** [comment]
Otherwise, https://github.com/Katello/katello/pull/9446 looks pretty ready (besides a small comment-nit), so I would expect this to go in today, and then we can merge this PR and follow up with a katello update to the deps there (can't be in the same PR)

**Review by evgeni — APPROVED** [review]

**Pull request merged**
